CardanoSharp.Wallet.Test/HDWallet/DerivationTests.cs

```diff
@@ -0,0 +1,258 @@
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var prv, var pub) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var master = testKey.Derive();
+            var purpose = master.Derive(PurposeType.Shelley);
+            var coin = purpose.Derive(CoinType.Ada);
+            var account = coin.Derive(0);
+            var role = account.Derive(RoleType.ExternalChain);
+            var index = role.Derive(0);
+
+            if (depth == 1) AssertDerivedKeys(prv, pub, purpose);
+            if (depth == 2) AssertDerivedKeys(prv, pub, coin);
+            if (depth == 3) AssertDerivedKeys(prv, pub, account);
+            if (depth == 4) AssertDerivedKeys(prv, pub, role);
+            if (depth == 5) AssertDerivedKeys(prv, pub, index);
+        }
+
+        [Fact]
+        public void ImplicitPathDerivationTest()
+        {
+            var path = "m/1852'/1815'/0'/0/0";
+
+            // Arrange
+            var mnemonic = new KeyService().Restore(__mnemonic);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var paymentPrv1, var paymentPub1) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var derivation = testKey.Derive(RoleType.ExternalChain)      // implicit IRoleNodeDerivation
+                                    .Derive(0);                          // IIndexNodeDerivation
+            var derivation2 = testKey.Derive(RoleType.ExternalChain, 0); // implicit IIndexNodeDerivation
+            var derivation3 = testKey.DeriveAccount(0)                   // implicit IAccountNodeDerivation
+                                    .Derive(RoleType.ExternalChain)      // implicit IRoleNodeDerivation
+                                    .Derive(0);                          // IIndexNodeDerivation
+
+            // Assert
+            AssertDerivedKeys(paymentPrv1, paymentPub1, derivation);
+            AssertDerivedKeys(paymentPrv1, paymentPub1, derivation2);
+            AssertDerivedKeys(paymentPrv1, paymentPub1, derivation3);
+        }
+
+        /// <summary>
+        /// Use Case:
+        /// I'm writing a light client (similar to Yoroi, Nami, C64Minter, etc). 
+        /// User enters in the Mnemonic and Spending Password. 
+        /// I derive down to the Account level and get the Public Key. 
+        /// I use the Spending Password to 2-Way encrypt the Private Key and then store both the encrypted Private Key and the plain Public Key.
+        /// </summary>
+        [Fact]
+        public void AccountKeyDerivation()
+        {
+            var path = WalletPath.Parse("m/1852'/1815'/0'");
+
+            // User enters in the Mnemonic and Spending Password. 
+            var mnemonic = _keyService.Generate(24);
+            PrivateKey rootKey = mnemonic.GetRootKey("password");
```

Comment:
Please note, this is not the spending password. This is impacts the keys being generated via the mnemonic. the spending password is solely responsible for Encrypting/Decrypting the Private Key